History

robertDurst opened the pull request:
## Changes
* Consolidated table to only two columns
* Redesigned table to be cleaner, more consistent, less of an "information vomit"
* Restructured the code format of history table row. Now each case in switch statement creates a details object which fits nicely into table row template. This allows for much nicer code readability. **Detailed description commented at top of file.**
* Added hover effect for assets in the history table rows
* Fixed filter functionality

## Design
Design of asset hover effect placement may need to be changed. For me this seemed like logical placement, right above the hovered asset name. 

## Screenshots

### Full Table
<img width="1439" alt="screen shot 2018-01-03 at 12 46 18 pm" src="https://user-images.githubusercontent.com/16689974/34539111-23be2e6e-f084-11e7-93e5-1db690b2b444.png">


### Trade Row
<img width="980" alt="screen shot 2018-01-03 at 12 43 33 pm" src="https://user-images.githubusercontent.com/16689974/34539068-f4a12faa-f083-11e7-93ad-5307bb357afd.png">

### Account Row
<img width="978" alt="screen shot 2018-01-03 at 12 43 50 pm" src="https://user-images.githubusercontent.com/16689974/34539064-f110885e-f083-11e7-9d08-81481a06b01e.png">

### Signer Row
<img width="981" alt="screen shot 2018-01-03 at 12 43 58 pm" src="https://user-images.githubusercontent.com/16689974/34539061-ecb27aba-f083-11e7-9a93-517ed043b9ed.png">

### Trustline Row
<img width="980" alt="screen shot 2018-01-03 at 12 44 10 pm" src="https://user-images.githubusercontent.com/16689974/34539059-e867084a-f083-11e7-9e64-481a7f554d65.png">

### Hover Row
<img width="982" alt="screen shot 2018-01-03 at 12 43 40 pm" src="https://user-images.githubusercontent.com/16689974/34539076-fda4988a-f083-11e7-91be-5e9bcf65cd26.png">

irisli commented:
Looks much better. Can you clean up the extra dead code in the driver? Also, there is a lot of stuff in the history that is no longer used.

For the filtering, I think the state should just live in the react component. The driver doesn't need to know filtering in order to properly work

robertDurst commented:
>  Can you clean up the extra dead code in the driver?

Fixed, cleaned out all the functions related to how effect history data was previously loaded in both MagicSpoon and Driver.

> For the filtering, I think the state should just live in the react component.

Filtering now local state in the HistoryView component.

>Also, there is a lot of stuff in the history that is no longer used.

I read through a few times and it looks like the old version of loading is gone and unused code gone as well. Let me know if I missed anything.
